### PR TITLE
Adding documentation for /search/assets and /bulk/asset/delete

### PR DESCRIPTION
**Pull request by atangOSIsoft**
Adding documentation for /search/assets and /bulk/assset/delete

### CODE REVIEW

File: Content_Portal/Documentation/Assets/asset-search-api.md [hunk 1]
```diff
@@ -25,11 +25,50 @@ Assets can be searched on the following asset properties:
...
+`string tenantId`  
+The tenant identifier
+
+`string namespaceId`  
+The namespace identifier
+
+`string queryString`  
+The asset query string. Search strings are not case-sensitive.
+
+[Optional] `int skip`
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/asset-search-api.md:46]: Our convention is to lowercase 'optional' as in"
[optional] `int skip`

File: Content_Portal/Documentation/Assets/asset-search-api.md [hunk 2]
```diff
@@ -25,11 +25,50 @@ Assets can be searched on the following asset properties:
...
+The namespace identifier
+
+`string queryString`  
+The asset query string. Search strings are not case-sensitive.
+
+[Optional] `int skip`   
+An optional parameter representing the zero-based offset of the first asset to retrieve. If not specified, a default value of 0 is used.
+
+[Optional] `int count`   
+An optional parameter representing the maximum number of assets to retrieve. If not specified, a default value of 100 is used. This value must be between (including) 1 to 1000.
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/asset-search-api.md:50]: Restate the first sentence: An optional parameter, between 1 and 1000 (inclusive), that represents the maximum number of assets to retrieve.
Delete the last sentence.

File: Content_Portal/Documentation/Assets/asset-search-api.md [hunk 3]
```diff
@@ -25,11 +25,50 @@ Assets can be searched on the following asset properties:
...
+
+[Optional] `[id|name] [asc|desc|] orderby`  
+An optional parameter which returns assets ordered either by the asset `Id` or the asset `name`. In addition to the property, you can optionally specify either `asc` or `desc` to return the results in ascending or descending order. If not specified, the default is ascending order.
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/asset-search-api.md:53]: Rewrite sentence 2-3 as: Specify `asc` or `desc` to return the results in ascending or descending order. The default is ascending order.

File: Content_Portal/Documentation/Assets/asset-search-api.md [hunk 4]
```diff
@@ -25,11 +25,50 @@ Assets can be searched on the following asset properties:
...
 
+### Response 
+Returns an array of assets matching the search query and the total number of assets returned specified as Total-Count in the HTTP response header.
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/asset-search-api.md:56]: Returns an array of assets matching the search query. The total number of returned assets is specified as Total-Count in the HTTP response header.

File: Content_Portal/Documentation/Assets/asset-search-api.md [hunk 5]
```diff
@@ -25,11 +25,50 @@ Assets can be searched on the following asset properties:
...
+| Status Code | Body Type | Description |
+|--|--|--|
+| 200 OK | Asset list | A list of assets matching the search criteria. An empty array is returned if there are no matching assets. |
+| 204 OK | empty | No assets match your search criteria. |
+| 400 Bad Request | error | The request is not valid. See the response body for additional details. |
+| 403 Forbidden | error | You are not authorized to view the requested asset. |
+
+
+## `Search Matched Fields Asset` 
+This API will search all assets and return to you a list of asset ids with their matched fields indicated in the return. This API can be used for identifying exactly which field in the asset matched your query string.
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/asset-search-api.md:67]: Rewrite to say: Searches all assets and returns a list of asset IDs and their matched fields. Use this API to identify the field in the asset that matched your query string. 

I'm not sure I understand how this is being used.

File: Content_Portal/Documentation/Assets/asset-search-api.md [hunk 6]
```diff
@@ -45,14 +84,14 @@ The asset query string. Search strings are not case-sensitive.
 [optional] `int skip`   
 An optional parameter representing the zero-based offset of the first asset to retrieve. If not specified, a default value of 0 is used.
 
-[optional] `int count`   
-An optional parameter representing the maximum number of assets to retrieve. If not specified, a default value of 100 is used.
+[Optional] `int count`   
+An optional parameter representing the maximum number of assets to retrieve. If not specified, a default value of 100 is used. This value must be between (including) 1 to 1000.
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/asset-search-api.md:88]: The maximum number of retrieved assets  between 1 and 1000 (inclusive). If not specified, the default is 100.

File: Content_Portal/Documentation/Assets/asset-search-api.md [hunk 7]
```diff
@@ -45,14 +84,14 @@ The asset query string. Search strings are not case-sensitive.
...
 
 [optional] `[id|name] [asc|desc|] orderby`  
 An optional parameter which returns assets ordered either by the asset `Id` or the asset `name`. In addition to the property, you can optionally specify either `asc` or `desc` to return the results in ascending or descending order. If not specified, the default is ascending order.
 
 ### Response 
-The response includes a status code and an array of assets matching the search criteria.
+A list of asset ids with their matched fields indicated.
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/asset-search-api.md:94]: A list of asset IDs and their matched fields.

File: Content_Portal/Documentation/Assets/asset-search-api.md [hunk 8]
```diff
@@ -61,7 +100,56 @@ The response includes a status code and an array of assets matching the search c
...
 | 403 Forbidden | error | You are not authorized to view the requested asset. |
 
-### Example Asset Queries
+#### Example response body
+Below is a response when query string is "Name:*Tracer*".
+
+`Count` represents the number of matched assets in the given query.
+
+Also returned is a list of `Results`. Each result contain
+    - The `MatchProperties` list contains a list of match property objects. Each match property object contains which field matched as well as the value of that field.
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/asset-search-api.md:109]: The `MatchProperties` list contains a list of matched property objects. Each matched property object contains the matched fields and their values.

File: Content_Portal/Documentation/Assets/asset-search-api.md [hunk 9]
```diff
@@ -61,7 +100,56 @@ The response includes a status code and an array of assets matching the search c
...
+    - `Score` which indicates the relevancy of the match. The higher the score, the more relevant this asset is to your query.
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/asset-search-api.md:110]: `Score` indicates the relevancy of the match.

File: Content_Portal/Documentation/Assets/asset-search-api.md [hunk 10]
```diff
@@ -61,7 +100,56 @@ The response includes a status code and an array of assets matching the search c
...
+    - `Id` the id of the matching asset.
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/asset-search-api.md:111]: `Id` Id of the matched asset

File: Content_Portal/Documentation/Assets/asset-search-api.md [hunk 11]
```diff
@@ -61,7 +100,56 @@ The response includes a status code and an array of assets matching the search c
...
+    - `TypeId` the asset type id of the asset. This may be null if the asset does not reference an asset type.
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/asset-search-api.md:112]: `TypeId` asset type Id of the asset. This is null if the asset does not reference an asset type.

File: Content_Portal/Documentation/Assets/asset-search-api.md [hunk 12]
```diff
@@ -61,7 +100,56 @@ The response includes a status code and an array of assets matching the search c
...
+    - `Name` the asset name.
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/asset-search-api.md:113]: `Name` asset name

File: Content_Portal/Documentation/Assets/asset-search-api.md [hunk 13]
```diff
@@ -61,7 +100,56 @@ The response includes a status code and an array of assets matching the search c
...
+    - `Description` the asset description.
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/asset-search-api.md:114]: `Description` asset description

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 14]
```diff
@@ -106,7 +106,7 @@ The namespace identifier
 An optional parameter representing the zero-based offset of the first asset to retrieve. If not specified, a default value of 0 is used.
 
 [optional] `int count`   
-An optional parameter representing the maximum number of assets to retrieve. If not specified, a default value of 100 is used. 
+An optional parameter representing the maximum number of assets to retrieve. If not specified, a default value of 100 is used. This value must be between (including) 1 to 100.
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:109]: An optional parameter, between 1 and 1000 (inclusive), that represents the maximum number of retrieved assets. If not specified, the default is 100.

File: Content_Portal/Documentation/Assets/asset-type.md [hunk 15]
```diff
@@ -89,7 +89,7 @@ The namespace identifier
 Maximum number of asset types to retrieve. If unspecified, the default (100) is used.
 
 [Optional] `int count`  
-An optional parameter representing the maximum number of assets to retrieve. If not specified, the default value of 100 is used.
+An optional parameter representing the maximum number of assets to retrieve. If not specified, the default value of 100 is used. This value must be between (including) 1 to 1000.
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/asset-type.md:92]: An optional parameter, between 1 and 1000 (inclusive), that represents the maximum number of retrieved assets. If not specified, the default is 100.

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 16]
```diff
@@ -301,4 +299,37 @@ The response includes a status code and a body.
 | 400 Bad Request           | error     | The request is not valid. The response includes which asset failed validation checks. See the response body for additional details.      |
 | 403 Forbidden             | error     | You are not authorized to delete this asset.       |
 | 404 Not Found             | error     | The asset with the specified `Id` could not be found.  |
-| 412 Pre-Condition Failed  | error     | The asset failed to update because the If-Match condition failed.  |
\ No newline at end of file
+| 412 Pre-Condition Failed  | error     | The asset failed to update because the If-Match condition failed.  |
+
+## `Delete Asset (bulk)`
+
+Delete all assets matching specified `Id`s. This API is used for deleting multiple assets in a given API call. Note that a maximum of 1000 assets can be deleted in a single call.
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:306]: Delete all assets matching the specified `Id`s. Use this API to delete up to a maximum of 1000 assets in one API call.

File: Content_Portal/Documentation/Assets/assets-api.md [hunk 17]
```diff
@@ -301,4 +299,37 @@ The response includes a status code and a body.
...
+A list of asset `Id`s.
+
+### Response 
+
+The response includes a status code and a body.
+
+| Status Code               | Body Type | Description                                     |
+| ------------------------- | --------- | ----------------------------------------------- |
+| 204 No Content            | none  | The assets with the specified `Id`s are deleted.                              |
+| 207 Multi-Status | partial success | Array of errors. Assets that didn't encounter errors would be deleted. |
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/assets-api.md:333]: Assets that did not encounter errors are deleted.

**Review by lasato-OSI — CHANGES_REQUESTED** [review]
Hi @atangOSIsoft ,
Please see my changes below. If you approve the changes, I will make the changes to the files.
thanks,
Laureen

File: Content_Portal/Documentation/Assets/asset-search-api.md [hunk 18]
```diff
@@ -61,7 +100,56 @@ The response includes a status code and an array of assets matching the search c
...
 
-### Example Asset Queries
+#### Example response body
+Below is a response when query string is "Name:*Tracer*".
+
+`Count` represents the number of matched assets in the given query.
+
+Also returned is a list of `Results`. Each result contains  
+- The `MatchProperties` list contains a list of match property objects. Each match property object contains which field matched as well as the value of that field.
+- `Score` which indicates the relevancy of the match. The higher the score, the more relevant this asset is to your query.
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/asset-search-api.md:110]: Score - number that indicates the relevancy of the match.

File: Content_Portal/Documentation/Assets/asset-search-api.md [hunk 19]
```diff
@@ -61,7 +100,56 @@ The response includes a status code and an array of assets matching the search c
...
+- `Id` the id of the matching asset.
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/asset-search-api.md:111]: Id - Id of the matched asset

File: Content_Portal/Documentation/Assets/asset-search-api.md [hunk 20]
```diff
@@ -61,7 +100,56 @@ The response includes a status code and an array of assets matching the search c
...
+- `TypeId` the asset type id of the asset. This may be null if the asset does not reference an asset type.
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/asset-search-api.md:112]: Type Id - the asset's asset type Id. This is null if the asset does not reference an asset type. 

Will it always be null or only some times?

File: Content_Portal/Documentation/Assets/asset-search-api.md [hunk 21]
```diff
@@ -61,7 +100,56 @@ The response includes a status code and an array of assets matching the search c
...
+- `Name` the asset name.
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/asset-search-api.md:113]: Name - asset name

File: Content_Portal/Documentation/Assets/asset-search-api.md [hunk 22]
```diff
@@ -61,7 +100,56 @@ The response includes a status code and an array of assets matching the search c
...
+- `Description` the asset description.
```
Comment (lasato-OSI) [inline Content_Portal/Documentation/Assets/asset-search-api.md:114]: Description - asset description

File: Content_Portal/Documentation/Assets/asset-search-api.md (hunk 20 above)
Comment (atangOSIsoft) [reply Content_Portal/Documentation/Assets/asset-search-api.md]: The typeId will only be null if the asset does not contains an asset type.

File: Content_Portal/Documentation/Assets/asset-search-api.md (hunk 4 above)
Comment (atangOSIsoft) [reply Content_Portal/Documentation/Assets/asset-search-api.md:56]: Thanks!

**Comment by atangOSIsoft** [comment]
> Hi @atangOSIsoft ,
> Please see my changes below. If you approve the changes, I will make the changes to the files.
> thanks,
> Laureen

Hi Laureen, the suggested changes look great to me. Thanks!

**Review by lasato-OSI — APPROVED** [review]

**Comment by lasato-OSI** [comment]
Adding documentation for /search/assets and /bulk/asset/delete.

**Pull request merged**